src/record_syscall.cc

```diff
@@ -4818,14 +4820,20 @@ static void rec_process_syscall_arch(RecordTask* t,
   }
 }
 
-static void rec_process_syscall_internal(RecordTask* t,
+/* N.B.: `arch` is the the architecture of the syscall, which may be different
+         from the architecture of the call (e.g. x86_64 may invoke x86 syscalls)
+*/
+static void rec_process_syscall_internal(RecordTask* t, SupportedArch arch,
                                          TaskSyscallState& syscall_state) {
-  RR_ARCH_FUNCTION(rec_process_syscall_arch, t->arch(), t, syscall_state)
+  RR_ARCH_FUNCTION(rec_process_syscall_arch, arch, t, syscall_state)
 }
 
 void rec_process_syscall(RecordTask* t) {
   auto& syscall_state = *syscall_state_property.get(*t);
-  rec_process_syscall_internal(t, syscall_state);
+  SupportedArch arch = t->ev().arch();
+  if (arch != t->arch())
+    LOG(warn) << "Cross architecture syscall detected. Support is best effort";
```

Comment:
I'm not sure it's really a great idea to print these on every syscall. Maybe set a global flag so this only gets printed once per rr run or something?